refactor/src/Cage.py

```diff
@@ -418,13 +426,36 @@ def do_mating_events(self):
                 df_list[fc].loc[df_list[fc].index.isin(dams),'mate_resistanceT1'] = \
                 df_list[fc].loc[df_list[fc].index.isin(sires),'resistanceT1'].values
         """
-        pass
 
-    def create_offspring(self):
+        # for now only consider a sensible number of lice that can mate
+        return min([10, self.lice_population["L5f"], self.lice_population["L5m"]])
+
+    def get_num_eggs(self, cur_month) -> int:
         """
-        TODO
+        Get the number of new eggs
+        :param cur_month the current month (to compute the temperature)
+        :returns the number of eggs produced
         """
 
+        # See Aldrin et al. 2017, §2.2.6
+        # TODO: in Aldrin matings are not taken into account. This may change
+        female_population = self.lice_population['L5f']
+        age_distrib = self.get_stage_ages_distrib("L5f")
+        density_rate = female_population * age_distrib / self.num_fish
+        # density_rate = mated_females / self.num_fish
+        age_range = np.arange(1, len(age_distrib) + 1)
+
+        density_factor = np.ones_like(density_rate) - np.exp(-self.cfg.reproduction_density_dependence * density_rate)
+        ave_temp = self.farm.year_temperatures[cur_month - 1]
+        temperature_factor = self.cfg.delta_m10["L0"] * (10 / ave_temp) ** self.cfg.delta_p["L0"]
```

Comment:
Is 10 a constant from Aldrin here?

Reply:
delta_m10 means median development factor based on a fixed temperature of 10 degrees.